iroha_logger/src/lib.rs

```diff
@@ -2,37 +2,104 @@
 
 use std::sync::atomic::{AtomicBool, Ordering};
 
+use async_std::sync::Receiver;
+use layer::LevelFilter;
 use once_cell::sync::Lazy;
+use telemetry::TelemetryFilter;
+use telemetry::{Telemetry, TelemetryLayer};
 pub use tracing::instrument as log;
+use tracing::subscriber::set_global_default;
 pub use tracing::Instrument;
 pub use tracing::Level;
 pub use tracing::{debug, error, info, trace, warn};
 pub use tracing::{debug_span, error_span, info_span, trace_span, warn_span};
 pub use tracing_futures::Instrument as InstrumentFutures;
-use tracing_subscriber::filter::LevelFilter;
-use tracing_subscriber::layer::SubscriberExt;
+
+pub mod layer;
+pub mod telemetry;
 
 static LOGGER_SET: Lazy<AtomicBool> = Lazy::new(|| AtomicBool::new(false));
 
 /// Initializes `Logger` with given `LoggerConfiguration`.
 /// After the initialization `log` macros will print with the use of this `Logger`.
-pub fn init(configuration: config::LoggerConfiguration) {
+pub fn init(configuration: config::LoggerConfiguration) -> Option<Receiver<Telemetry>> {
     if LOGGER_SET
         .compare_exchange(false, true, Ordering::AcqRel, Ordering::Relaxed)
-        .is_ok()
+        .is_err()
     {
-        #[cfg(not(test))]
-        let subscriber = tracing_subscriber::fmt()
-            .compact()
-            .finish()
-            .with(LevelFilter::from(configuration.max_log_level));
-        #[cfg(test)]
-        let subscriber = tracing_subscriber::fmt()
-            .finish()
-            .with(LevelFilter::from(configuration.max_log_level));
-        #[allow(clippy::expect_used)]
-        tracing::subscriber::set_global_default(subscriber).expect("Failed to init logger");
+        return None;
     }
+    let fmt = tracing_subscriber::fmt().compact().finish();
+    let level = configuration.max_log_level.into();
+    let (subscriber, receiver) =
+        TelemetryLayer::new::<1000>(TelemetryFilter::new(LevelFilter::new(level, fmt)));
+
+    #[allow(clippy::expect_used)]
+    set_global_default(subscriber).expect("Failed to init logger");
+    Some(receiver)
+}
+
+/// Macro for sending telemetry info
+#[macro_export]
+macro_rules! telemetry {
```

Comment:
How the telemetry intended to be send? Via `telemetry!` macros or 'iroha_logger!' macros?Or both possible?

Reply:
I asked because everywhere `iroha_logger` used

Reply:
Both possible but preferable telemetry macros